WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java

```diff
@@ -472,9 +472,12 @@ else if (mPost.getDate_created_gmt() > 0)
             mPost.setLocation(mPostLocation);
         }
 
+        if (mCategories != null) {
+            mPost.setJSONCategories(new JSONArray(mCategories));
+        }
```

Comment:
Should the post categories be cleared/nulled when `mCategories` is null?

Also, there are a number of other places in this fragment where `mCategories` is accessed without a null check ([for example](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java#L307)). Should these be addressed as well?